submissions/zalevskyi/friends-app/app.js

```diff
@@ -0,0 +1,187 @@
+const USERS_DATA = []
+const USERS_QUANTITY = 30
+const RESULTS = document.getElementById('results')
+const SORT = document.getElementById('sort')
+const FILTER = document.getElementById('filter')
+const REQUEST_PARAMETERS = {}
+let connectionAttemps = 1
+
+function connectToServer() {
+  RESULTS.innerHTML = `<div class='connection_message'>Please wait. Connecting to server. Attemp: ${connectionAttemps}</div>`
+  connectionAttemps++
+  fetchUserData(connectToServer)
+}
+function fetchUserData(onError) {
+  let apiUrl = 'https://randomuser.me/api/1.3/'
+  let fields = ['name', 'gender', 'location', 'dob', 'phone', 'picture']
+  fetch(`${apiUrl}?results=${USERS_QUANTITY}&inc=${fields.join(',')}`, {
+    headers: { 'Content-Type': 'application/json' },
+  })
+    .then((response) => {
+      if (!response.ok) {
+        throw Error(response.statusText)
+      }
+      return response
+    })
+    .then((response) => response.json())
+    .then((responseJSON) => {
+      pushUserData(responseJSON.results)
+      showResults()
+    })
+    .catch((error) => onError(error))
+}
+function pushUserData(randomUserResults) {
+  randomUserResults.forEach((user) =>
+    USERS_DATA.push({
+      name: user.name.first,
+      surname: user.name.last,
+      gender: user.gender,
+      age: user.dob.age,
+      country: user.location.country,
+      city: user.location.city,
+      phone: user.phone,
+      image: user.picture.large,
+    })
+  )
+}
+function showResults() {
+  RESULTS.innerHTML = getUserDataToShow(USERS_DATA, REQUEST_PARAMETERS)
+    .map((user) => getUserCard(user))
+    .join('')
+}
+function getUserDataToShow(userData, parameters) {
+  if (parameters.sort) {
+    return sort(filter(userData, parameters), parameters.sort, parameters.order)
+  } else {
+    return filter(userData, parameters)
+  }
+}
+function sort(userData, fieldToSort, order) {
+  if (fieldToSort === 'name') {
+    return sortFullName(userData, 'name', order)
```

Comment:
Passing the string key is a bad idea - you can misprint and you will have a bug. Instead, your sorting function should accept concrete values from two objects - friend A and friend B.